src/ORM/Query.php

```diff
@@ -194,6 +195,30 @@ public function select($fields = [], $overwrite = false)
         return parent::select($fields, $overwrite);
     }
 
+    /**
+     * Select all the columns associated to $table except the $excludedFields.
+     * Excluded fields should not be aliased names.
+     *
+     *
+     * @param Table|Association $table The table to use to get an array of columns
+     * @param array $excludedFields The un-aliased column names you do not want selected from $table
+     * @param bool $overwrite Whether to reset/remove previous selected fields
+     * @return Query
+     */
+    public function selectAllExcept($table, array $excludedFields, $overwrite = false)
+    {
+        if ($table instanceof Association) {
+            $table = $table->getTarget();
+        }
+        $aliasedFields = [];
+        if ($table instanceof Table) {
```

Comment:
Maybe if the target is not a Table throw an Exception instead?

Reply:
Does it have to be a concrete table? Don't we usually try to program against interfaces for SOLID?

Reply:
Only `Table` class has `getSchema()` method so checking for `RepositoryInterface()` isn't feasible.

This also makes me wonder if it's right to have a method on `Query` with takes an `Association` or `Table` as argument.

Reply:
Maybe it should take a some Interface that both implement?

Reply:
Lorenzo recommended that the first argument be Table so there would be no confusion about what list of columns the user was selecting. (See original thread https://github.com/cakephp/cakephp/pull/11025).  

For now, I've left the parameters alone.  We could type hint Table and make the caller know to caller know that they have to call getTarget() on their association. The good news is the doc blocks will tell me what to pass.

It now throws an exception if something other than a table or association is passed.
Updated documentation and tests as requested.